cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -103,16 +107,25 @@ static void _sc_cgroup_v1_attach_pid(sc_device_cgroup *self, pid_t pid) {
     sc_dprintf(self->v1.fds.cgroup_procs_fd, "%i\n", getpid());
 }
 
-sc_device_cgroup *sc_device_cgroup_new(const char *security_tag) {
+
+static void sc_device_cgroup_close(sc_device_cgroup *self);
+
+sc_device_cgroup *sc_device_cgroup_new(const char *security_tag, int flags) {
     sc_device_cgroup *self = calloc(1, sizeof(sc_device_cgroup));
     if (self == NULL) {
         die("cannot allocate device cgroup wrapper");
     }
     self->is_v2 = sc_cgroup_is_v2();
     self->security_tag = sc_strdup(security_tag);
 
+    int ret = 0;
     if (!self->is_v2) {
-        _sc_cgroup_v1_init(self);
+        ret = _sc_cgroup_v1_init(self, flags);
+	}
+
+    if (ret < 0) {
+        sc_device_cgroup_close(self);
```

Comment:
I missed this in my last batch of tweaks. I'll think about it a push a patch.

Reply:
I think this one is correct. `_sc_cgroup_v1_init` returns -1 only if the group did not exist before, which means `SC_DEVICE_CGROUP_FROM_EXISTING` must have been passed, as otherwise s-c would die(). As a precaution, close is called, such that any state data (of v1 and the wrapper) set up so far would be cleaned properly. I guess one could argue that nothing should have been allocated if the whole thing wasn't successful. There will be a bit more changes in the area with #10575 so I suggest we revisit this when I open the actual PR with device cgroup v2.

Reply:
Ok let's revisit once there is more context.